File: src/taller/taller3/FixedSubwayCard.java
package taller.taller3;

public class FixedSubwayCard extends SubwayCard {
    private int rides;

    FixedSubwayCard(SubwayCentral sc, int rides) {
        super(sc);
        this.rides = rides;
    }

    @Override
    protected boolean canRide() {
        return rides > 0;
    }

    @Override
    public void ride() {
        if(canRide()) {
            rides--;
        }
        else {
            System.out.println("No se puede viajar");
        }
    }




}

File: POO-Ruby/guias/guia8/ej6/fixed_subway_card.rb
# frozen_string_literal: true
require_relative 'subway_card'

class FixedSubwayCard < SubwayCard
  def initialize(subway_central, rides)
    @subway_central = subway_central
    raise 'Invalid rides' unless rides.is_a?(Integer) && rides.positive?
    @rides = rides
  end

  private def ride?
    @rides.positive?
  end

  def ride
    raise 'No rides left' unless ride?
    @rides -= 1
  end

  def to_s
    "Fixed #{super} [Rides: #{@rides}]"
  end
end
